src/Reexport.jl

```diff
@@ -1,28 +1,44 @@
 module Reexport
 
-macro reexport(ex)
-    isa(ex, Expr) && (ex.head == :module ||
-                      ex.head == :using ||
-                      (ex.head == :toplevel &&
-                       all(e->isa(e, Expr) && e.head == :using, ex.args))) ||
+macro reexport(ex::Expr)
+    esc(reexport(__module__, ex))
+end
+
+reexport(m::Module, l::LineNumberNode) = l
+
+function reexport(m::Module, ex::Expr)
+    # unpack any macros
+    ex = macroexpand(m, ex)
+    # recursively unpack any blocks
+    if ex.head == :block
+        return Expr(:block, map(e -> reexport(m, e), ex.args)...)
+    end
+
+    Meta.isexpr(ex, [:module, :using, :import]) ||
```

Comment:
Note that `Meta.isexpr` does a check for `ex isa Expr`, but we know by the signature of this method that `ex` is an `Expr`; it's constrained to be so. I think this could instead be `ex.head in [...]` and likewise in a few other places.